ppafm/io.py

```diff
@@ -471,224 +471,304 @@ def saveXSF(fname, data, lvec=None, dd=None, head=XSF_HEAD_DEFAULT, verbose=1 ):
+    head.append("g98_3D_unknown \n")
+    head.append("DATAGRID_3D_g98Cube \n")
+    FF*=Hartree2eV
+    return FF,lvec, nDim, head
 
 # ================ WSxM output
 
 def saveWSxM_2D(name_file, data, Xs, Ys):
-	tmp_data=data.flatten()
-	out_data=np.zeros((len(tmp_data),3))
-	out_data[:,0]=Xs.flatten()
-	out_data[:,1]=Ys.flatten()
-	out_data[:,2]=tmp_data	#.copy()
-	f=open(name_file,'w')
-	print("WSxM file copyright Nanotec Electronica", file=f)
-	print("WSxM ASCII XYZ file", file=f)
-	print("X[A]  Y[A]  df[Hz]", file=f)
-	print("", file=f)
-	np.savetxt(f, out_data)
-	f.close()
+    tmp_data=data.flatten()
+    out_data=np.zeros((len(tmp_data),3))
+    out_data[:,0]=Xs.flatten()
+    out_data[:,1]=Ys.flatten()
+    out_data[:,2]=tmp_data    #.copy()
+    f=open(name_file,'w')
+    print("WSxM file copyright Nanotec Electronica", file=f)
+    print("WSxM ASCII XYZ file", file=f)
+    print("X[A]  Y[A]  df[Hz]", file=f)
+    print("", file=f)
+    np.savetxt(f, out_data)
+    f.close()
 
 def saveWSxM_3D( prefix, data, extent, slices=None ):
-	nDim=np.shape(data)
-	if slices is None:
-		slices=list(range( nDim[0]))
-	xs=np.linspace( extent[0], extent[1], nDim[2] )
-	ys=np.linspace( extent[2], extent[3], nDim[1] )
-	Xs, Ys = np.meshgrid(xs,ys)
-	for i in slices:
-		print("slice no: ", i)
-		fname = prefix+'_%03d.xyz' %i
-		saveWSxM_2D(fname, data[i], Xs, Ys)
+    nDim=np.shape(data)
+    if slices is None:
+        slices=list(range( nDim[0]))
+    xs=np.linspace( extent[0], extent[1], nDim[2] )
+    ys=np.linspace( extent[2], extent[3], nDim[1] )
+    Xs, Ys = np.meshgrid(xs,ys)
+    for i in slices:
+        print("slice no: ", i)
+        fname = prefix+'_%03d.xyz' %i
+        saveWSxM_2D(fname, data[i], Xs, Ys)
 
 # ================ Npy
 
-def saveNpy(fname, data, lvec , head=None):
-	np.save(fname+'.npy', data)
-	np.save(fname+'_vec.npy',lvec)
+def saveNpy(fname, data, lvec , atomic_info):
+    '''
+    Function for saving scalar grid data, together with its lattice_vector and information about original atoms and the original lattice vector (lvec0) in numpy format
+
+        Arguments:
+            fname: str. Path to file.
+            data: np.ndarray of shape (n_z, n_y, n_x) with scallar data
+            lvec: np.ndarray of shape (4, 3). Lattice vector of the data
+            atomic_info: tuple of shape (2). First part is [e, x, y, z] of atoms, the second is lvec of the atoms from the original geometry file, named as lvec0;
+    '''
+    np.savez(fname+'.npz', data=data, lvec=lvec, atoms=atomic_info[0], lvec0=atomic_info[1])
+
 
 def loadNpy(fname):
-	data = np.load(fname+'.npy')
-	lvec = np.load(fname+'_vec.npy')
-	return data.copy(), lvec;	#necessary for being 'C_CONTINUOS'
+    '''
+    Function for loading scalar grid data, together with its lattice_vector and information about original atoms and the original lattice vector (lvec0) in numpy format
+
+    Arguments:
+        fname: str. name of the npz file.
```

Comment:
The docstrings for this function and the others dealing with the numpy format should probably mention that the file name should be without the extension.